js/views/userPageVw.js

```diff
@@ -322,15 +318,29 @@ module.exports = baseVw.extend({
     });
   },
 
-  bindModalCloseHandler: function(e) {
-    if (this.modalCloseHandler) return;
-
-    this.modalCloseHandler = function() {
-      window.history.back();
+  loadingStatus: function() {
+    var config = {
+      promise: this.loadingDeferred.promise(),
```

Comment:
`promise` is the one mandatory property to opt into the Page Connect modal functionality (i.e. have the page connecting modal show when we're trying to connect to the page).

You are free to pass in a `cancel` function that will be called if the user clicks the `cancel` button on the modal. It's a place you could do in any cleanup, like aborting any XHR's.